ls1intum/Artemis
# Limit number of locked submissions per tutor

### Checklist

### Motivation and Context


To prevent that tutors misuse the locking mechanism we limit the number of simultaneously locked submissions, i.e. unfinished assessments, per student. The number of submissions that can be locked by each tutor gets limited to 10 per course.

### Description

The submission lock limit works as follows:
- When the user opens the tutor exercise dashboard and the lock limit is reached, a red alert message is shown in the dashboard and he cannot start a new assessment (first screenshot). The alert message stays in the exercise dashboard until the user finished or canceled an open assessment.
- When the user tries to access the assessment editor directly (via URL) to start a new assessment and the lock limit is reached, the user gets redirected to the exercise dashboard with the same alert message as above.
- In modeling exercises: when the user clicks on "Assess Next Submission" in the modeling assessment editor and the lock limit is reached, he gets redirected to the exercise dashboard with the same alert message as above.

### Steps for Testing


1. Log in to ArTEMiS as a tutor
2. Start assessments for 10 different (modeling or text) submissions
3. Go to the exercise dashboard
4. See alert and notice that you cannot start a new assessment

### Screenshots


Tutor Exercise Dashboard (lock limit reached)
![Exercise dashboard with alert](https://user-images.githubusercontent.com/5919138/59641976-9939c280-9163-11e9-953d-17ce41bba302.JPG)
